readthedocs/v3/views.py

```diff
@@ -0,0 +1,239 @@
+from django.shortcuts import get_object_or_404
+from django.utils.safestring import mark_safe
+import django_filters.rest_framework as filters
+from rest_framework.authentication import SessionAuthentication, TokenAuthentication
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAdminUser
+from rest_framework.renderers import JSONRenderer, BrowsableAPIRenderer
+from rest_framework.response import Response
+from rest_framework.throttling import UserRateThrottle, AnonRateThrottle
+from rest_framework.mixins import ListModelMixin, RetrieveModelMixin, UpdateModelMixin, CreateModelMixin
+from rest_framework.viewsets import GenericViewSet
+from rest_framework_extensions.mixins import NestedViewSetMixin
+from rest_flex_fields.views import FlexFieldsMixin
+
+from readthedocs.core.utils import trigger_build
+from readthedocs.builds.models import Version, Build
+from readthedocs.projects.models import Project
+from rest_framework.metadata import SimpleMetadata
+from .filters import ProjectFilter, VersionFilter, BuildFilter
+from .serializers import ProjectSerializer, VersionSerializer, VersionUpdateSerializer, BuildSerializer
+
+
+class APIv3Settings:
+
+    authentication_classes = (SessionAuthentication, TokenAuthentication)
+    permission_classes = (IsAdminUser,)
+    renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
+    throttle_classes = (UserRateThrottle, AnonRateThrottle)
+    filter_backends = (filters.DjangoFilterBackend,)
+    metadata_class = SimpleMetadata
+
+
+class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, FlexFieldsMixin, ListModelMixin, RetrieveModelMixin, GenericViewSet):
+
+    """
+    Endpoints related to ``Project`` objects.
+
+    * Listing objects.
+    * Detailed object.
+
+    Retrieving only needed data using ``?fields=`` URL attribute is allowed.
+
+    ### Filters
+
+    Allowed via URL attributes:
+
+    * slug
+    * slug__contains
+    * name
+    * name__contains
+
+    ### Expandable fields
+
+    There are some fields that are not returned by default because they are
+    expensive to calculate. Although, they are available for those cases where
+    they are needed.
+
+    Allowed via ``?expand=`` URL attribue:
+
+    * users
+    * active_versions
+    * active_versions.last_build
+    * active_versions.last_build.confg
+
+
+    ### Examples:
+
+    * List my projects: ``/api/v3/projects/``
+    * Filter list: ``/api/v3/projects/?name__contains=test``
+    * Retrieve only needed data: ``/api/v3/projects/?fields=slug,created``
+    * Retrieve specific project: ``/api/v3/projects/{project_slug}/``
+    * Expand required fields: ``/api/v3/projects/{project_slug}/?expand=active_versions``
+    * Translations of a projects: ``/api/v3/projects/{project_slug}/translations/``
```

Comment:
```suggestion
    * Translations of a project: ``/api/v3/projects/{project_slug}/translations/``
```

Same for the ones below